docs/sdk/src/polkadot_sdk/xcm.rs

```diff
@@ -1,5 +1,38 @@
 //! # XCM
 //!
-//! @KiChjang @franciscoaguirre
-//! TODO: RFCs, xcm-spec, the future of the repo, minimal example perhaps, forward to where actual
-//! docs are hosted. https://github.com/paritytech/polkadot-sdk-docs/issues/58
+//! XCM, or Cross-Consensus Messaging, is a **language** to communicate **intentions** between **consensus systems**.
+//!
+//! ## Overview
+//!
+//! XCM is a language agnostic standard, whose specification lives in the [xcm format repo](https://github.com/paritytech/xcm-format).
+//!
+//! It enables different consensus systems to communicate with each other in an expressive manner.
+//! Consensus systems include blockchains, smart contracts, and any other state machine that achieves consensus in some way.
+//!
+//! XCM is based on a virtual machine, the XCVM.
+//! An XCM program is a series of instructions, which get executed one after the other by the virtual machine.
+//! These instructions aim to encompass all major things users typically do in consensus systems.
+//! There are instructions on asset transferring, teleporting, locking, among others.
+//! New instructions are added via the [RFC process](https://github.com/paritytech/xcm-format/blob/master/proposals/0032-process.md).
+//!
+//! ## In Polkadot SDK
+//!
+//! The Polkadot SDK allows for easily deploying own blockchains from scratch, all very customizable.
+//! This makes dealing with all these blockchains cumbersome.
+//! XCM exists both as a standard for doing similar things in many chains, and as a language to send message between them.
+//!
+//! ## Implementation
+//!
+//! A ready-to-use Rust implementation lives in the [polkadot-sdk repo](https://github.com/paritytech/polkadot-sdk/tree/master/polkadot/xcm),
+//! but will be moved to its own repo in the future.
+//!
+//! Its main components are:
+//! - `src`: the definition of the basic types and instructions
+//! - `xcm-executor`: an implementation of the virtual machine to execute instructions
```

Comment:
link to the crate/type, eg https://paritytech.github.io/polkadot-sdk/master/xcm_simulator/struct.XcmExecutor.html?